src/D2L.Bmx/Aws/AwsClient.cs

```diff
@@ -18,69 +14,23 @@ internal class AwsClient : IAwsClient {
 		_stsClient = stsClient;
 	}
 
-	AwsRoleState IAwsClient.GetRoles( string encodedSaml ) {
-		var samlToken = ParseSamlToken( encodedSaml );
-		var roleNodes = samlToken.SelectNodes( "//*[@Name=\"https://aws.amazon.com/SAML/Attributes/Role\"]/*" );
-
-		var roles = new List<AwsRole>();
-
-		if( roleNodes is not null ) {
-			foreach( XmlElement roleNode in roleNodes ) {
-				// SAML has value: <principal-arn>, <role-arn>
-				// The last part of the role-arn is a human readable name
-				string[] nodeContents = roleNode.InnerText.Split( "," );
-
-				roles.Add( new AwsRole(
-					RoleName: nodeContents[1].Split( "/" )[1],
-					PrincipalArn: nodeContents[0],
-					RoleArn: nodeContents[1]
-					) );
-			}
-		} else {
-			throw new BmxException( "Failed to retrieve roles" );
-		}
-		return new AwsRoleState( roles, encodedSaml );
-	}
-
 	async Task<AwsCredentials> IAwsClient.GetTokensAsync(
-		AwsRoleState state,
-		string selectedRole,
-		int durationInMinutes ) {
-		var role = state.AwsRoles
-			.Find( role => string.Equals( role.RoleName, selectedRole, StringComparison.OrdinalIgnoreCase ) );
```

Comment:
Similarly this is also internal processing that should be done by the caller not the client method.
There's also no need for the client call to know about all the available roles.